filebeat/crawler/prospector_log.go

```diff
@@ -251,14 +254,13 @@ func (p *ProspectorLog) checkExistingFile(h *harvester.Harvester, newFile *input
 		// We only need to keep it for the remainder of this iteration then we can assume it was deleted and forget about it
 		p.missingFiles[h.Path] = oldFile.FileInfo
 
-	} else if h.Stat.Finished() && oldFile.FileInfo.ModTime() != h.Stat.Fileinfo.ModTime() {
+	} else if h.State.Finished() {
```

Comment:
Yes, it is. The reason is that during the new setup phase, there is now a state sent to a harvester. This can mean, that a file was rotated but now new content was added. This hits h.State.Finished() and then an updated state must be sent.

If a file was renamed, the modified time didn't change so it was removed. TBH I'm not to 100% sure in which way it required before but I think it was to prevent that for a file two harvesters are started. 

My plan is to merge the functions checkExistingFile and checkNewFile into one and heavily reduce the complexity here. This should soon be possible as the state handling is now moved outside these functions. Bear with me for a few more PR's ;-) Step by step.